### PR TITLE
Add list of supported monorepo managers

### LINKED ISSUE #7757: [Monorepo Detection] Add monorepo-managers.ts
This file will mimic the structure of `vercel/frameworks/frameworks.ts` in the sense that it defines the list of supported monorepo managers.  These are the ones were targeting for the initial launch,

- Turborepo

```ts
export const monorepoManagers = [
  {
    name: "Turborepo",
    slug: "turbo",
    detectors: {
      every: [
        {
          path: 'turbo.json'
        },
      ],
    },
  },
]
```

### PR DESCRIPTION
**Pull request by agadzik**
This file will mimic the structure of `vercel/frameworks/frameworks.ts` in the sense that it defines the list of supported monorepo managers.  These are the ones were targeting for the initial launch,

- Turborepo

```ts
export const monorepoManagers = [
  {
    name: "Turborepo",
    slug: "turbo",
    detectors: {
      every: [
        {
          path: 'turbo.json'
        },
      ],
    },
  },
]
```

### Related Issues

> Closes https://github.com/vercel/vercel/issues/7757

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [x] Issue from task tracker has a link to this PR

### CODE REVIEW

File: packages/build-utils/src/workspaces/workspace-managers.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ import type { Framework } from '@vercel/frameworks';
  * This list is designed to work with the @see {@link detectFramework} function.
  *
  * @example
- *   import { workspaceManagers as frameworkList } from '@vercel/build-utils/workspaces'
+ *   import { workspaceManagers as frameworkList } from '@vercel/build-utils'
```
Comment (chloetedder) [inline packages/build-utils/src/workspaces/workspace-managers.ts:9]:
```suggestion
 *   import { workspaceManagers as workspaceManagerList } from '@vercel/build-utils'
```

Comment (agadzik) [reply packages/build-utils/src/workspaces/workspace-managers.ts:9]: I think we're ok with my changes as we're showing that we can rename `workspaceManagers` to `frameworkList` and pass that into `detectFramework` 🙌

File: packages/build-utils/src/monorepos/monorepo-managers.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+      outputDirectory: {
+        value: '',
+        placeholder: '',
+      },
+    },
+    getOutputDirName: () => Promise.resolve(''),
+  },
+];
+
+export default monorepoManagers;
```
Comment (chloetedder) [inline packages/build-utils/src/monorepos/monorepo-managers.ts:36]: Wondering if we should add these to the index file to export just from `@vercel/build-utils`?

Comment (agadzik) [reply packages/build-utils/src/monorepos/monorepo-managers.ts:36]: opps - good catch!  Added

File: packages/build-utils/src/monorepos/monorepo-managers.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+ * @example
+ *   import { monorepoManagers as frameworkList } from '@vercel/build-utils'
+ *   import { detectFramework } from '@vercel/build-utils'
+ *
+ *   const fs = new GitDetectorFilesystem(...)
+ *   detectFramwork({ fs, frameworkList }) // returns the 'slug' field if detected, otherwise null
+ *
+ * @todo Will be used by the detect-eligible-projects API endpoint for a given git url.
+ */
+export const monorepoManagers: Array<Framework> = [
```
Comment (styfle) [inline packages/build-utils/src/monorepos/monorepo-managers.ts]: Should this code live in `@vercel/frameworks` instead?

I'm wondering if we need a `@vercel/monorepos` pkg instead for all this new code?

Thoughts @TooTallNate ?

File: packages/build-utils/src/monorepos/monorepo-managers.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    name: 'Turborepo',
+    slug: 'turbo',
+    detectors: {
+      every: [
+        {
+          path: 'turbo.json',
+        },
+      ],
+    },
+    // unused props - needed for typescript
```
Comment (styfle) [inline packages/build-utils/src/monorepos/monorepo-managers.ts]: This seems like the wrong solution. Perhaps we need a new type

File: packages/build-utils/src/monorepos/monorepo-managers.ts (hunk 3 above)
Comment (nathanhammond) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: Realistically there are three things at play: frameworks (lol no way to list them all), package managers (npm, pnpm, yarn, berry), package manager managers (ecosystem integration tools) (Turborepo, lerna, lerna-lite, [wireit](https://github.com/google/wireit), ...)

Note that I'd *maybe* shy away from the term "monorepo" as we don't consider Turborepo to be constrained by monorepo.

File: packages/build-utils/src/monorepos/monorepo-managers.ts [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ * @todo Will be used by the detect-eligible-projects API endpoint for a given git url.
+ */
+export const monorepoManagers: Array<Framework> = [
+  {
+    name: 'Turborepo',
+    slug: 'turbo',
+    detectors: {
+      every: [
+        {
+          path: 'turbo.json',
```
Comment (nathanhammond) [inline packages/build-utils/src/monorepos/monorepo-managers.ts:25]: This isn't guaranteed. Previously our config was at the `turbo` key inside of `$ROOT/package.json`.

File: packages/build-utils/src/monorepos/monorepo-managers.ts (hunk 4 above)
Comment (agadzik) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: I suspect we'll find an use for them in the future, but for now they're unused.  These changes were based on @Ethan-Arrowood's comments here https://github.com/vercel/vercel/pull/7737#discussion_r865340977

File: packages/build-utils/src/monorepos/monorepo-managers.ts (hunk 3 above)
Comment (agadzik) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: Happy to move code if needed.  Our thoughts were to just add them to `build-utils` at first, then refactor if needed

Comment (agadzik) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: could call the new package (if needed) `@vercel/repo-utils`

Comment (nathanhammond) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: Yeah, this is just a naming/organization question; I have no strong opinions other than avoiding the word "monorepo." And maybe even "repo" ... `@vercel/project-utils` maybe?

**Review by nathanhammond — APPROVED** [review]
From the perspective of Turborepo this is good to go.

File: packages/build-utils/src/monorepos/monorepo-managers.ts (hunk 4 above)
Comment (styfle) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: The suggestion there mentions Omit

Let use something like this:

```
Array<Omit<Framework, 'logo' | 'settings' | 'getOutputDirName'>>
```

Comment (agadzik) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: image

File: packages/build-utils/src/monorepos/monorepo-managers.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+ * This list is designed to work with the @see {@link detectFramework} function.
+ *
+ * @example
+ *   import { monorepoManagers as frameworkList } from '@vercel/build-utils'
+ *   import { detectFramework } from '@vercel/build-utils'
+ *
+ *   const fs = new GitDetectorFilesystem(...)
+ *   detectFramwork({ fs, frameworkList }) // returns the 'slug' field if detected, otherwise null
+ *
+ * @todo Will be used by the detect-eligible-projects API endpoint for a given git url.
```
Comment (EndangeredMassa) [inline packages/build-utils/src/monorepos/monorepo-managers.ts]: Will this `@todo` be removed after https://github.com/vercel/api/issues/12179 is done? If so, can we add a note on that issue to make sure this is removed later?

(I'm still learning how different teams are tracking tasks. Thanks!)

Comment (agadzik) [reply packages/build-utils/src/monorepos/monorepo-managers.ts]: oh shoot - ill just remove this! good catch

**Review by EndangeredMassa — APPROVED** [review]
Given the current state of the `@vercel/frameworks` package, I think this is a reasonable approach.

That said, the use of `Omit` to remove required keys from the `Framework` type seems brittle (because an addition to `Framework` may require a change here) and incomplete (because the optional keys still exist on the new resulting type, but there's not clear intent to use those.

Not in this PR, but in the near future: I think we should refactor frameworks to better accommodate this kind of use, colocate appropriate logic (based on concerns brought up in this PR discussion), rename terms and repos (based on concerns brought up in this PR discussion), and allow for different versions to have different values ([identified here](https://github.com/vercel/vercel/pull/7808#issuecomment-1130333998)). Not all of that would have to be done at once, of course. I can write up an RFC, unless this really isn't a big deal right now.

File: packages/build-utils/src/monorepos/monorepo-managers.ts [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+ *
+ * @example
+ *   import { monorepoManagers as frameworkList } from '@vercel/build-utils'
+ *   import { detectFramework } from '@vercel/build-utils'
+ *
+ *   const fs = new GitDetectorFilesystem(...)
+ *   detectFramwork({ fs, frameworkList }) // returns the 'slug' field if detected, otherwise null
+ *
+ */
+export const monorepoManagers: Array<
```
Comment (styfle) [inline packages/build-utils/src/monorepos/monorepo-managers.ts:16]: In hindsight, this should probably be

```ts
Array<Pick<Framework, 'name' | 'slug' | 'detectors'>>
```

**Pull request merged**